### PR TITLE
Fix: Fixed issue with resolving paths containing dots

### LINKED ISSUE #9808: Bug: Files doesn't fully resolve path containing ".."
### Description

When opening a path such as `%appdata%\..\LocalLow`, Files doesn't fully resolve the path, instead expanding the `appdata` but keeping a `..\LocalLow` after it.

### Steps To Reproduce

1. Go to the "bar"
2. Type in `%appdata%\..\LocalLow`
3. Files doesn't fully resolve the path

**Requirements**
- Files properly resolves paths such as `AppData\Roaming\..\LocalLow` into `AppData\LocalLow`.

### Files Version

Version 2.3.7.0

### Windows Version

Windows 11 2H22 25182.1000

### Relevant Assets/Logs

image

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #9808
  Related #11575

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?

### CODE REVIEW

File: src/Files.App/Filesystem/StorageFileHelpers/StorageFileExtensions.cs [hunk 1]
```diff
@@ -145,29 +140,10 @@ public static List<PathBoxItem> GetDirectoryPathComponents(string value)
...
-			{
-				path = path.Replace("%localappdata%", CommonPaths.LocalAppDataPath, StringComparison.OrdinalIgnoreCase);
-			}
-			if (path.Contains("%homepath%", StringComparison.OrdinalIgnoreCase))
-			{
-				path = path.Replace("%homepath%", CommonPaths.HomePath, StringComparison.OrdinalIgnoreCase);
-			}
-			return Environment.ExpandEnvironmentVariables(path);
+			var withoutEnvirnment = GetPathWithoutEnvironmentVariable(path);
+			return ResolvePath(path, isFtp);
```
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/Filesystem/StorageFileHelpers/StorageFileExtensions.cs]: Thank you, I'm blind

File: src/Files.App/Filesystem/StorageFileHelpers/StorageFileExtensions.cs [hunk 2]
```diff
@@ -327,5 +301,104 @@ private static PathBoxItem GetPathItem(string component, string path)
...
+				return "Home";
+
+			var pathBuilder = new StringBuilder(path);
+			var lastPathIndex = path.Length - 1;
+			var separatorChar = isFtp || path.Contains('/', StringComparison.Ordinal) ? '/' : '\\';
+			var rootIndex = isFtp ? FtpHelpers.GetRootIndex(path) + 1 : path.IndexOf($":{separatorChar}", StringComparison.Ordinal) + 2;
+
+			for (int i = 0, lastIndex = 0; i < pathBuilder.Length; i++)
+			{
+				if (pathBuilder[i] is '?' ||
```
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/Filesystem/StorageFileHelpers/StorageFileExtensions.cs]: Done

**Comment by yaira2** [comment]
@Josh65-2201 can you test the behavior and verify it's working as expected?

**Review by yaira2 — APPROVED** [review]
Thank you!

**Pull request merged**